### PR TITLE
Fix some warnings and treat compiler warnings as errors

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
     <IsPackable>false</IsPackable>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```
Comment (bgavrilMS) [inline Directory.Build.props:14]: This is for compiler warnings, which are different than Roslyn analyzers. 

@jennyf19 - consider enabling this in M.I.W as well.

Comment (jennyf19) [reply Directory.Build.props:14]: How is line 14 different from line 3? should there be a condition included or one removed?

Comment (trwalke) [reply Directory.Build.props:14]: So we are now treating all warnings as errors for the entire project? Will the Xamarin projects also be affected by this?

Comment (bgavrilMS) [reply Directory.Build.props:14]: I'm struggling with this, there are too many warnings from NuGet / msbuild that get treated as errors :( in the CI build, but not locally!

But if you don't have any MSB or NU warning, I think it's the way to go.

**Pull request merged**